common/src/commonTest/kotlin/dev/inmo/micro_utils/common/MapDiffUtilsTests.kt

```diff
@@ -0,0 +1,377 @@
+            diff.changed
+        )
+    }
+
+    @Test
+    fun testApplyMapDiffRemovedWithCompareFun() {
+        val wasCalled = compareFunWasCalled
+        val oldMap = originalMap.toMutableMap()
+        val diff = oldMap.applyDiff(newMapRemoved.first, compareFun(wasCalled))
+
+        assertTrue(wasCalled[0])
+        assertEquals(
+            newMapRemoved.second,
+            diff.removed
+        )
+    }
+    @Test
+    fun testApplyMapDiffAddedWithCompareFun() {
+        val wasCalled = compareFunWasCalled
+        val oldMap = originalMap.toMutableMap()
+        val diff = oldMap.applyDiff(newMapAdded.first, compareFun(wasCalled))
+
+        assertTrue(wasCalled[0])
+        assertEquals(
+            newMapAdded.second,
+            diff.added
+        )
+    }
+
+    @Test
+    fun testApplyMapDiffChangedWithCompareFun() {
+        val wasCalled = compareFunWasCalled
+        val oldMap = originalMap.toMutableMap()
+        val diff = oldMap.applyDiff(newMapChangedComparable.first, compareFun(wasCalled))
+
+        assertTrue(wasCalled[0])
+        assertEquals(
+            newMapChangedComparable.second,
+            diff.changed
+        )
+    }
+
+    @Test
+    fun testMapDiffMixed() {
+        val oldMap = originalMap.toMutableMap()
+        val diff = oldMap.applyDiff(newMapMixed.first)
+
+        assertEquals(
+            newMapMixed.second.removed,
+            diff.removed
+        )
+        assertEquals(
+            newMapMixed.second.changed,
+            diff.changed
+        )
+        assertEquals(
+            newMapMixed.second.added,
+            diff.added
+        )
+    }
+
+    @Test
+    fun testApplyMapDiffMixed() {
+        val oldMap = originalMap.toMutableMap()
+        val diff = oldMap.applyDiff(newMapMixed.first)
+
+        assertEquals(
+            newMapMixed.second.removed,
+            diff.removed
+        )
+        assertEquals(
+            newMapMixed.second.changed,
+            diff.changed
+        )
+        assertEquals(
+            newMapMixed.second.added,
+            diff.added
+        )
+    }
+
+    @Test
+    fun testMapDiffWorksWithEqualValuesStrictComparison() {
+        val oldMap = mutableMapOf<String, Int>()
+        val newMap = mutableMapOf<String, Int>()
+
+        for (i in 0 until 10) {
+            val value = Random.nextInt()
+            oldMap[i.toString()] = value
+            newMap[i.toString()] = value
+        }
+
+        val mapDiff = oldMap.diff(newMap, true)
+
+        val mergedValues = oldMap.keys.intersect(newMap.keys).associateWith {
+            oldMap[it] to newMap[it]
+        }
+
+        assertEquals(
+            mergedValues,
+            mapDiff.changed,
+        )
+        assertEquals(
+            emptyMap(),
+            mapDiff.removed
+        )
+        assertEquals(
+            emptyMap(),
+            mapDiff.added
+        )
+    }
+
+    @Test
+    fun testApplyMapDiffWorksWithEqualValuesStrictComparison() {
```

Comment:
Here you should check that:

1. Values from old map (before changes) presented in diffs with strict (`===`) comparison
2. Values from old map absent in the oldMap after applyDiff (with strict check too)
3. Values in oldMap after applyDiff strictly (`===`) equal to the values from newMap by the same keys
4.